vowpalwabbit/csoaa.cc

```diff
@@ -129,7 +145,69 @@ void predict_or_learn(csoaa& c, single_learner& base, example& ec)
   ec.pred.multiclass = prediction;
 }
 
-void finish_example(vw& all, csoaa&, example& ec) { COST_SENSITIVE::finish_example(all, ec); }
+template <bool probabilities>
+void finish_example(vw& all, csoaa& c, example& ec)
+{
+  if (probabilities)
+  {
+    // === Compute multiclass_log_loss
+    // TODO:
```

Comment:
Would be nice if we could either resolve the todos or determine if they are relevant before copying them over here